pkg/apis/manager/v1alpha3/api.proto

```diff
@@ -516,13 +516,14 @@ message GetObservationLogReply {
 }
 
 message GetSuggestionsRequest {
-    string experiment_name = 1;
-    string algorithm_name = 2;
+    Experiment experiment = 1;
+    repeated Trial trials = 2; // all completed trials owned by the experiment.
     int32 request_number = 3; ///The number of Suggestion you request at one time. When you set 3 to request_number, you can get three Suggestions at one time.
 }
 
 message GetSuggestionsReply {
-    repeated Trial trials = 1;
+    repeated Trial trials = 1; // trials should be created in the next run.
+    AlgorithmSpec algorithm = 2;
```

Comment:
How do we intend to use AlgorithmSpec in the reply?

Reply:
It is used to support hyperband. If the algorithm wants to maintain some states using algorithm settings, I think we should support it in the current version. In the future, we should remove it.